dev/io.openliberty.microprofile.telemetry.1.0.internal/src/io/openliberty/microprofile/telemetry10/internal/rest/TelemetryClientFilter.java

```diff
@@ -49,48 +51,75 @@ public class TelemetryClientFilter extends AbstractTelemetryClientFilter impleme
 
     private static final TraceComponent tc = Tr.register(TelemetryClientFilter.class);
 
-    private final Instrumenter<ClientRequestContext, ClientResponseContext> instrumenter;
+    private Instrumenter<ClientRequestContext, ClientResponseContext> instrumenter;
+    private volatile boolean lazyCreate = false;
+    private final AtomicReference<Instrumenter<ClientRequestContext, ClientResponseContext>> lazyInstrumenter = new AtomicReference<>();
 
     private final String configString = "otel.span.client.";
 
     private static final NetClientAttributesGetterImpl NET_CLIENT_ATTRIBUTES_GETTER = new NetClientAttributesGetterImpl();
     private static final HttpClientAttributesGetterImpl HTTP_CLIENT_ATTRIBUTES_GETTER = new HttpClientAttributesGetterImpl();
 
     TelemetryClientFilter() {
+        if (instrumenter == null) {
+            if (!CheckpointPhase.getPhase().restored()) {
+                lazyCreate = true;
```

Comment:
We should only set `lazyCreate` to true if `openTelemetryInfo.getEnabled() && !AgentDetection.isAgentActive()` is true.  Otherwise it looks like we will continually try to create it over and over.  This applies to all cases here.

There is also a method on this class:

```java
    @Override
    public boolean isEnabled() {
        return instrumenter != null;
    }
```
This method will need to be updated to `return instrumenter != null || lazyCreate;`

That can only be done if you ensure `lazyCreate` is set to true when `openTelemetryInfo.getEnabled() && !AgentDetection.isAgentActive()` is true like I mention above

Reply:
Fixed isEnabled() for checkpoint and restore scenario.